### PR TITLE
Reorganize and add Trait::Traced to the debugging programs page

### PR DESCRIPTION
**Pull request by Kaiepi**
## The problem
- The debugging programs page is a bit disorganized. More emphasis is placed on ecosystem modules than core features, and what paragraphs pertain to which section isn't entirely clear to me at first glance.
- Albeit a somewhat new module, `Trait::Traced` offers broader support for tracing than other ecosystem modules out there (to my knowledge). I may be a bit biased since it's my own module though 😉.

## Solution provided
- Reorganize the debugging programs page.
- Add `Trait::Traced` to the ecosystem modules section.

### CODE REVIEW

File: doc/Programs/01-debugging.pod6 [hunk 1]
```diff
@@ -107,6 +79,43 @@ a string:
...
+line|https://github.com/rakudo/rakudo/wiki/Running-rakudo-from-the-command-line>
+for more information.
+
+=head1 Ecosystem debugging modules
+
+There are at least two useful debuggers and two tracers available for Rakudo
+(the Raku compiler).  For more information on these modules, please refer to
+their documentation.
+
+Historically other modules have existed and others are likely to be
```
Comment (JJ) [inline doc/Programs/01-debugging.pod6]: This kinda interrupts the flow. You talk about the modules, but only start enumerating them later. Maybe put this apart as a footnote? It sounds like a footnote, really.

File: doc/Programs/01-debugging.pod6 [hunk 2]
```diff
@@ -107,6 +79,43 @@ a string:
...
+written in the future. Please check the L<Raku
+Modules|https://modules.raku.org/> website for more modules like these.
+
+=head2 L<Debugger::UI::CommandLine|https://modules.raku.org/repo/Debugger::UI::CommandLine>
```
Comment (JJ) [inline doc/Programs/01-debugging.pod6]: Maybe surround it with the traditional C<>?

File: doc/Programs/01-debugging.pod6 (hunk 1 above)
Comment (Kaiepi) [reply doc/Programs/01-debugging.pod6]: How are footnotes usually formatted?

File: doc/Programs/01-debugging.pod6 (hunk 2 above)
Comment (Kaiepi) [reply doc/Programs/01-debugging.pod6]: Sounds good

File: doc/Programs/01-debugging.pod6 (hunk 1 above)
Comment (JJ) [reply doc/Programs/01-debugging.pod6]: Just wrap them in N<>, and put them where the footnote will be referred to.

**Pull request merged**